models/iaf_cond_alpha_mc_fixedca.cpp

```diff
@@ -0,0 +1,853 @@
+  const long_t from,
+  const long_t to )
+{
+
+
+  assert(
+    to >= 0 && ( delay ) from < kernel().connection_manager.get_min_delay() );
+
+  assert( from < to );
+
+  for ( long_t lag = from; lag < to; ++lag )
+  {
+
+    double t = 0.0;
+
+    // numerical integration with adaptive step size control:
+    // ------------------------------------------------------
+    // gsl_odeiv_evolve_apply performs only a single numerical
+    // integration step, starting from t and bounded by step;
+    // the while-loop ensures integration over the whole simulation
+    // step (0, step] if more than one integration step is needed due
+    // to a small integration step size;
+    // note that (t+IntegrationStep > step) leads to integration over
+    // (t, step] and afterwards setting t to step, but it does not
+    // enforce setting IntegrationStep to step-t; this is of advantage
+    // for a consistent and efficient integration across subsequent
+    // simulation intervals
+    while ( t < B_.step_ )
+    {
+      const int status = gsl_odeiv_evolve_apply( B_.e_,
+        B_.c_,
+        B_.s_,
+        &B_.sys_,             // system of ODE
+        &t,                   // from t
+        B_.step_,             // to t <= step
+        &B_.IntegrationStep_, // integration step size
+        S_.y_ );              // neuronal state
+
+      if ( status != GSL_SUCCESS )
+        throw GSLSolverFailure( get_name(), status );
+    }
+
+    S_.th_ += ( S_.th_ - P_.V_th ) * ( V_.AdaptThStep_ );
+
+    // add incoming spikes at end of interval
+    // exploit here that spike buffers are compartment for compartment,
+    // alternating between excitatory and inhibitory
+    for ( size_t n = 0; n < NCOMP; ++n )
+    {
+      S_.y_[ n * State_::STATE_VEC_COMPS + State_::DG_EXC ] +=
+        B_.spikes_[ 2 * n ].get_value( lag ) * V_.PSConInit_E_[ n ];
+      S_.y_[ n * State_::STATE_VEC_COMPS + State_::DG_INH ] +=
+        B_.spikes_[ 2 * n + 1 ].get_value( lag ) * V_.PSConInit_I_[ n ];
+    }
+
+    // refractoriness and spiking
+    // exploit here that plain offset enum value V_M indexes soma V_M
+    if ( S_.r_ )
+    { // neuron is absolute refractory
+      --S_.r_;
+      if ( S_.r_ == int( 0.5 * V_.RefractoryCounts_ ) )
+      {
+        S_.y_[ PROX * State_::STATE_VEC_COMPS + State_::DI_AP ] +=
+          P_.amp_currAP[ PROX ] * V_.PSConInit_AP_[ PROX ];
+      }
+      if ( S_.r_ == 0 )
+      {
+        if ( P_.reset_on_spike )
+          S_.y_[ State_::V_M ] = P_.V_reset;
+        S_.y_[ DIST * State_::STATE_VEC_COMPS + State_::DI_AP ] +=
+          P_.amp_currAP[ DIST ] * V_.PSConInit_AP_[ DIST ];
+      }
+    }
+    else if ( S_.y_[ State_::V_M ] >= S_.th_ )
+    { // neuron fires spike
+      S_.r_ = V_.RefractoryCounts_;
+      S_.y_[ State_::V_M ] = P_.V_max;
+      S_.th_ += P_.jump_Th;
+      P_.g_L[ SOMA ] = P_.t_L[ SOMA ];
+      P_.g_L[ PROX ] = P_.t_L[ PROX ];
+      P_.g_L[ DIST ] = P_.t_L[ DIST ];
```

Comment:
Why are you changing *parameters* in `update()`? The elements of `P_` should only be changed in `Parameters_::set()`!